modules/bbmap/index/main.nf

```diff
@@ -0,0 +1,41 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+def VERSION = '38.92'       // There seems to be no way to get the version out of the bbmap.sh script, only a line with "Last modified" and a date
+
+process BBMAP_INDEX {
+    tag "$fasta"
+    label 'process_long'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::bbmap=38.92" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bbmap:38.92--he522d1c_0"
+    } else {
+        container "quay.io/biocontainers/bbmap:38.92--he522d1c_0"
+    }
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path('ref')  , emit: db
```

Comment:
```suggestion
    tuple val(meta), path('ref')  , emit: db
```

Maybe just `index` for the channel? To keep consistency with the module name?

Reply:
I can't see any difference, did you mean `emit: index`?

Reply:
Yes sorry. Early morning reviews not the best idea :laughing:

Reply:
:-) I just pushed a separate commit with this.